main.cpp

```diff
@@ -56,64 +56,78 @@ Create a branch named Part1
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value;
+    std::string name;
+    
+    T(int v, const char* name)    //1 . I'm wondering about casting from char* to std::string here? It only shows the firs character in the console
+    {
+        value = v;    //2
+        this->name = *name;    //3
```

Comment:
std::string can be initialized with `const char*`.  don't dereference or initialize in-class.  use member initializer list to init it.  Also, change your arg name to not overshadow the member variable.

- [ ] rename arg

- [ ] init members via member initializer list.

Reply:
Got it, made these changes